relayer/relayer/src/trader/mod.rs

```diff
@@ -0,0 +1,303 @@
+                        debug!("{whoami_azero} has A0 balance too low for bridging");
+                        continue;
+                    }
+
+                    info!("{azero_available_for_swap} A0 above the safe limit will be swapped.");
+
+                    let min_weth_amount_out = match router
+                        .get_amounts_out(azero_signed_connection.as_connection(), azero_available_for_swap, &swap_path)
+                        .await
+                    {
+                        Ok(amounts) => {
+                            debug!("Amounts out {amounts:?}.");
+
+                            match amounts.last() {
+                                Some(amount) => amount.saturating_mul(100 - SLIPPAGE_PERCENT).saturating_div(100),
+                                None => {
+                                    warn!("Query to `calculate_amounts_out` returned an empty result.");
+                                    continue;
+                                }
+                            }
+                        },
+
+                        Err(why) => {
+                            warn!("Could not `get_amounts_out`: {why:?}");
+                            continue;
+                        }
+                    };
+
+                    let now = SystemTime::now()
+                        .duration_since(UNIX_EPOCH)
+                        .expect("unix timestamp")
+                        .as_millis();
+
+                    info!("Requesting a swap of {azero_available_for_swap} pA0 to at least {min_weth_amount_out} Azero ETH.");
+
+                    if let Err(why) = router
+                        .swap_exact_native_for_tokens(
+                            &azero_signed_connection,
+                            azero_available_for_swap,
+                            min_weth_amount_out,
+                            &swap_path,
+                            whoami_azero.clone(),
+                            now.saturating_add(HOUR_IN_MILLIS.into()) as u64, // within one hour
+                        )
+                        .await
+                    {
+                        warn!("Could not perform swap: {why:?}.");
+                        continue;
+                    }
+
+                    // check azero Eth balance
+                    let azero_eth_balance = match azero_ether
+                        .balance_of(azero_signed_connection.as_connection(), whoami_azero.clone())
+                        .await {
+                            Ok(balance) => balance,
+                            Err(why) => {
+                                warn!("Error when querying for Azero ETH balance: {why:?}.");
+                                continue;
+                            },
+                        };
+
+                    // Never bridge less than 0.1 ETH
+                    if azero_eth_balance > ONE_ETHER / 10 {
```

Comment:
Suggestion: If you want to comment, I'd suggest commenting about _why_ we don't want to bridge less, not just the fact that we don't.

```suggestion
                    if azero_eth_balance > BRIDGING_THRESHOLD {
```